runtime/parachains/src/paras/mod.rs

```diff
@@ -291,8 +291,15 @@ pub struct ParaGenesisArgs {
 	pub genesis_head: HeadData,
 	/// The initial validation code to use.
 	pub validation_code: ValidationCode,
-	/// True if parachain, false if parathread.
-	pub parachain: bool,
+	/// Parachain or Parathread.
+	pub paratype: ParaType,
+}
+
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum ParaType {
+	ParaChain,
```

Comment:
Nit: style similar to `ParaLifecycle`
```suggestion
	Parachain,
```